xds/internal/balancer/clusterresolver/priority_test.go

```diff
@@ -1,834 +0,0 @@
-
-	addrs1 := <-cc.NewSubConnAddrsCh
-	if got, want := addrs1[0].Addr, testEndpointAddrs[0]; got != want {
-		t.Fatalf("sc is created with addr %v, want %v", got, want)
-	}
-	sc1 := <-cc.NewSubConnCh
-
-	// p0 is ready.
-	edsb.UpdateSubConnState(sc1, balancer.SubConnState{ConnectivityState: connectivity.Connecting})
-	edsb.UpdateSubConnState(sc1, balancer.SubConnState{ConnectivityState: connectivity.Ready})
-
-	// Test roundrobin with only p0 subconns.
-	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
-	defer cancel()
-	if err := cc.WaitForRoundRobinPicker(ctx, sc1); err != nil {
-		t.Fatal(err)
-	}
-
-	// Add p2, it shouldn't cause any updates.
-	clab2 := xdstestutils.NewClusterLoadAssignmentBuilder(testClusterNames[0], nil)
-	clab2.AddLocality(testSubZones[0], 1, 0, testEndpointAddrs[:1], nil)
-	clab2.AddLocality(testSubZones[1], 1, 1, testEndpointAddrs[1:2], nil)
-	clab2.AddLocality(testSubZones[2], 1, 2, testEndpointAddrs[2:3], nil)
-	xdsC.InvokeWatchEDSCallback("", parseEDSRespProtoForTesting(clab2.Build()), nil)
-
-	select {
-	case <-cc.NewSubConnCh:
-		t.Fatalf("got unexpected new SubConn")
-	case <-cc.RemoveSubConnCh:
-		t.Fatalf("got unexpected remove SubConn")
-	case <-time.After(defaultTestShortTimeout):
-	}
-
-	select {
-	case p := <-cc.NewPickerCh:
-		// If we do get a new picker, ensure it is still a p1 picker.
-		if err := testutils.IsRoundRobin([]balancer.SubConn{sc1}, testutils.SubConnFromPicker(p)); err != nil {
-			t.Fatal(err)
-		}
-	default:
-		// No new picker; we were previously using p1 and should still be using
-		// p1, so this is okay.  No need to wait for defaultTestShortTimeout
-		// since we just waited immediately above.
-	}
-
-	// Remove p2, no updates.
-	clab3 := xdstestutils.NewClusterLoadAssignmentBuilder(testClusterNames[0], nil)
-	clab3.AddLocality(testSubZones[0], 1, 0, testEndpointAddrs[:1], nil)
-	clab3.AddLocality(testSubZones[1], 1, 1, testEndpointAddrs[1:2], nil)
-	xdsC.InvokeWatchEDSCallback("", parseEDSRespProtoForTesting(clab3.Build()), nil)
-
-	select {
-	case <-cc.NewSubConnCh:
-		t.Fatalf("got unexpected new SubConn")
-	case <-cc.RemoveSubConnCh:
-		t.Fatalf("got unexpected remove SubConn")
-	case <-time.After(defaultTestShortTimeout):
-	}
-
-	select {
-	case p := <-cc.NewPickerCh:
-		// If we do get a new picker, ensure it is still a p1 picker.
-		if err := testutils.IsRoundRobin([]balancer.SubConn{sc1}, testutils.SubConnFromPicker(p)); err != nil {
-			t.Fatal(err)
-		}
-	default:
-		// No new picker; we were previously using p1 and should still be using
-		// p1, so this is okay.  No need to wait for defaultTestShortTimeout
-		// since we just waited immediately above.
-	}
-
-}
-
-// Lower priority is used when higher priority is not ready.
-//
-// Init 0 and 1; 0 is up, use 0; 0 is down, 1 is up, use 1; add 2, use 1; 1 is
-// down, use 2; remove 2, use 1.
-func (s) TestEDSPriority_SwitchPriority(t *testing.T) {
```

Comment:
I know all these tests are a bit clunky, but can you make sure that all the tests deleted don't delete extra operations/assertions that are captured in /xds/internal/balancer/priority/balancer_test.go.

Reply:
From my reading of it, it does look like the tests in the priority package cover all scenarios. We definitely will have to rewrite them as e2e style tests. But I'm not planning to hike up that mountain until it becomes a necessity :). 

One small hill at a time.